tests/Doctrine/Tests/ORM/Functional/Ticket/DDC6303Test.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+/**
+ * @group DDC6303
+ */
+class DDC6303Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+        try {
+            $this->_schemaTool->createSchema(
+                [
+                $this->_em->getClassMetadata(DDC6303Contract::class),
+                $this->_em->getClassMetadata(DDC6303ContractA::class),
+                $this->_em->getClassMetadata(DDC6303ContractB::class)
+                ]
+            );
+        } catch (\Exception $ignored) {}
+    }
+    
+    public function testMixedTypeHydratedCorrectlyInJoinedInheritance()
+    {
+        $contractA = new DDC6303ContractA();
+        $contractAData = 'authorized';
+        $contractA->originalData = $contractAData;
+
+        $contractB = new DDC6303ContractB();
+        $contractBData = ['accepted', 'authorized'];
```

Comment:
Could you add comments for the reader, specifying that these two entities map `originalData` differently?